From e407f41c7c181218da79265ca90fdd3ad60bbd76 Mon Sep 17 00:00:00 2001
From: Manish Bhardwaj <mbhardwaj@nvidia.com>
Date: Wed, 26 May 2021 23:01:18 +0530
Subject: [PATCH] K5.10: fixes UBSAN related issue

This patch fixes below issues :-
1. UBSAN: signed-integer-overflow in /dvs/git/dirty/git-master_modular/out/
aarch64-arm64-tegra_defconfig-rt_patches-debug-extmod_ubuntu18.04_aarch64-extmod_linux_x86_64/
kernel/src-rt/kernel-5.10/kernel/locking/rwlock-rt.c:175:2
-2147483648 - 1073741824 cannot be represented in type 'int'
2. UBSAN: signed-integer-overflow in /dvs/git/dirty/git-master_modular/out/
aarch64-arm64-tegra_defconfig-rt_patches-debug-extmod_ubuntu18.04_aarch64-extmod_linux_x86_64/
kernel/src-rt/kernel-5.10/kernel/locking/rwsem-rt.c:221:2
-2147483648 - 1073741824 cannot be represented in type 'int'

Bug 200734434

Signed-off-by: Manish Bhardwaj <mbhardwaj@nvidia.com>
Change-Id: I423eb742707f02d7c2563829c80430946fce3db9
---
 include/linux/rwlock_types_rt.h | 4 ++--
 include/linux/rwsem-rt.h        | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/linux/rwlock_types_rt.h b/include/linux/rwlock_types_rt.h
index 3e9c037..8b4798e 100644
--- a/include/linux/rwlock_types_rt.h
+++ b/include/linux/rwlock_types_rt.h
@@ -33,8 +33,8 @@ struct rt_rw_lock {
 #endif
 };
 
-#define READER_BIAS	(1 << 31)
-#define WRITER_BIAS	(1 << 30)
+#define READER_BIAS	(1 << 30)
+#define WRITER_BIAS	(1 << 29)
 
 #define __RWLOCK_RT_INITIALIZER(name)					\
 {									\
diff --git a/include/linux/rwsem-rt.h b/include/linux/rwsem-rt.h
index fb4c442..dcb1b82 100644
--- a/include/linux/rwsem-rt.h
+++ b/include/linux/rwsem-rt.h
@@ -9,8 +9,8 @@
 #include <linux/rtmutex.h>
 #include <linux/swait.h>
 
-#define READER_BIAS		(1 << 31)
-#define WRITER_BIAS		(1 << 30)
+#define READER_BIAS		(1 << 30)
+#define WRITER_BIAS		(1 << 29)
 
 struct rw_semaphore {
 	atomic_t		readers;
-- 
2.7.4

